core/src/main/scala/coulomb/conversion/standard/standard.scala

```diff
@@ -50,16 +50,16 @@ inline given ctx_VC_Long[VF](using num: Integral[VF]): ValueConversion[VF, Long]
     new ValueConversion[VF, Long]:
         def apply(v: VF): Long = num.toLong(v)
 
-inline given ctx_VC_Long_tr[VF](using num: Fractional[VF], at: AllowTruncation): ValueConversion[VF, Long] =
-    new ValueConversion[VF, Long]:
+inline given ctx_TVC_Long[VF](using num: Fractional[VF]): TruncatingValueConversion[VF, Long] =
+    new TruncatingValueConversion[VF, Long]:
         def apply(v: VF): Long = num.toLong(v)
```

Comment:
This is a bikeshed for another issue/PR, but I'm concerned that `inline` combined with `new ...` is sub-optimal and can lead to a proliferation of anonymous classes in the jar. It depends on whether the compiler will automatically recognize this is a SAM and use `invokedynamic` instead (or. It might be better to implement this as a lambda e.g. `v => num.toLong(v)` to encourage this optimization.

I'm less familiar with Scala.js internals but this could be especially bad for JavaScript actually.

See https://github.com/typelevel/cats/pull/3871 for a discussion of this topic.

Reply:
Actually, it's not obvious to me why `inline` is needed here?

Reply:
Do not assume I'm using these _only_ when they are needed  :grin:

I'm still working out my intuitions about when `inline` and `transparent inline` are necessary.  Situations where a typeclass involved dependent typing seem unambiguous - they need `transparent inline`. Others are less obvious (to me).

Some places I am definitely using `inline` for efficiency gains, but other places it may be a bad trade

At any rate, if `inline` is problematic, I'm happy to clean all that up where it is not needed.  Having some decent unit test coverage should make this a bit easier now.  If something introduces a compile error, it ought to show up pretty fast.

Reply:
Yes, I would use `inline` where needed only for its compile-time semantics.

For runtime performance, both the JVM and the Scala.js linker backend do a fantastic job of inlining themselves.

We can always revisit `inline` opportunities in the future with a benchmark suite.

Reply:
We have had bad experiences with macro generated classes in the past. When defining boopickle generated serialized it was very easy to endup generating thousands of instances that would bloat the resulting js size. It wasn't really noticeable on the jvm thus it was quite hard to spot

Reply:
overall I expect scala-3 coulomb to be _better_ in this regard.  I am using metaprogramming in multiple places where I was using intermediate typeclasses, and so these intermediate typeclasses should no longer appear in byte code.  However, some of this may be offset by use of `transparent inline`, so it will definitely be helpful to remove any unimportant usages of `inline`